libnetwork/error.go

```diff
@@ -53,8 +53,8 @@ func (nnr NetworkNameError) Error() string {
 	return fmt.Sprintf("network with name %s already exists", string(nnr))
 }
 
-// Forbidden denotes the type of this error
-func (nnr NetworkNameError) Forbidden() {}
+// Conflict denotes the type of this error
+func (nnr NetworkNameError) Conflict() {}
```

Comment:
Oh, I got confused for a moment because of the `NetworkNameError` name. I wish it was called something like `NetworkExistsError`.

Reply:
Right, it's a bit confusing. I'll rename it.

Reply:
I was also wondering if the error still had to be exported, or if code could use `errdefs.IsConflict()` instead, or are there other situations where a `Conflict()` error can be produced for other reasons that should be treated differently?

It looks like some code for swarm networks uses the error (actually wondering if that means duplicate networks can still exist); https://github.com/moby/moby/blob/4b19b2f4babd577a11bd0c1c018e7263e51d49fa/daemon/cluster/executor/container/adapter.go#L213-L216

And I see it's used for creating the ingress network.. that part of the code looks a bit "iffy" though; we log the error as an _error_, but continue business as usual. If that code-path is "expected" (as in: user-created ingress network?) it should probably not be logged as an _error_. https://github.com/moby/moby/blob/4b19b2f4babd577a11bd0c1c018e7263e51d49fa/daemon/network.go#L211-L219

Reply:
> Oh, I got confused for a moment because of the `NetworkNameError` name. I wish it was called something like `NetworkExistsError`.

I see we use `nameConflictError` for containers, so we could use the same / similar; https://github.com/moby/moby/blob/4b19b2f4babd577a11bd0c1c018e7263e51d49fa/daemon/errors.go#L51-L60

Reply:
> I was also wondering if the error still had to be exported, or if code could use errdefs.IsConflict() instead, or are there other situations where a Conflict() error can be produced for other reasons that should be treated differently?
> I see we use nameConflictError for containers, so we could use the same / similar;

I'm not sure if there's other errors implement `Conflict` in this code path but IMO it's less robust to check for the generic `Conflict` interface rather than a specific error, and it doesn't communicate the real intent. If we replace `NetworkNameError` with an un-exported symbol or a "generic" error, it'll just make the code harder to follow.

> actually wondering if that means duplicate networks can still exist

I dug that rabbit hole a bit yesterday and I believe the answer is yes and no. Two `CreateNetwork` requests sent to two manager nodes could race against each other (ie. two networks stored in managers' MemDB). But since networks are lazily "provisioned" on a node (including managers) when a task is scheduled on that specific node, and given that this lazy provisioning logic just ignores when a `NetworkNameError` is returned (that's the code in `adapter.go`), then only a single network will be provisioned by Swarm.

Reply:
> and it doesn't communicate the real intent.

Yes, agreed. Was mostly thinking out loud here, but I agree.

We should probably look at replacing such cases with an `errors.Is()` / `errors.As()` just in case things are wrapped; definitely not for this PR though.